src/System.Security.Cryptography.Algorithms/tests/CryptoConfigTests.cs

```diff
@@ -234,6 +236,57 @@ public static void CreateFromName_CtorArguments()
             Assert.Equal("Hello", ctorObj.MyString);
         }
 
+        [Fact]
+        public static void EncodeOID_Validation()
+        {
+            Assert.Throws<ArgumentNullException>(() => CryptoConfig.EncodeOID(null));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID(string.Empty));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID("BAD.OID"));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID("1.2.BAD.OID"));
+            Assert.Throws<OverflowException>(() => CryptoConfig.EncodeOID("1." + uint.MaxValue));
+            CryptoConfig.EncodeOID("-1.2.-3"); // Negative values not checked
```

Comment:
That's... depressing.   It should probably be moved out to a test with "compat" in the name and checked for a stable output.